sys/include/net/nanocoap_sock.h

```diff
@@ -154,19 +154,99 @@ extern "C" {
  * @param[in]   path    remote path
  * @param[out]  buf     buffer to write response to
  * @param[in]   len     length of @p buffer
  *
  * @returns     length of response payload on success
  * @returns     <0 on error
  */
-ssize_t nanocoap_get(sock_udp_ep_t *remote, const char *path, uint8_t *buf,
+ssize_t nanocoap_get(sock_udp_t *sock, const char *path, void *buf,
                      size_t len);
 
+/**
+ * @brief    Performs a blockwise coap get request on a socket.
+ *
+ * This function will fetch the content of the specified resource path via
+ * block-wise-transfer. A coap_blockwise_cb_t will be called on each received
+ * block.
+ *
+ * @param[in]   sock       socket to use for the request
+ * @param[in]   path       pointer to source path
+ * @param[in]   blksize    sender suggested SZX for the COAP block request
+ * @param[in]   work_buf   Work buffer, must be `NANOCOAP_BLOCKWISE_BUF(blksize)` bytes
+ * @param[in]   callback   callback to be executed on each received block
+ * @param[in]   arg        optional function arguments
+ *
+ * @returns     -EINVAL    if an invalid url is provided
+ * @returns     -1         if failed to fetch the url content
+ * @returns      0         on success
+ */
+int nanocoap_get_blockwise(sock_udp_t *sock, const char *path,
+                           coap_blksize_t blksize, void *work_buf,
+                           coap_blockwise_cb_t callback, void *arg);
+
+/**
+ * @brief    Performs a blockwise coap get request to the specified url.
+ *
+ * This function will fetch the content of the specified resource path via
+ * block-wise-transfer. A coap_blockwise_cb_t will be called on each received
+ * block.
+ *
+ * @param[in]   url        Absolute URL pointer to source path (i.e. not containing a fragment identifier)
+ * @param[in]   blksize    sender suggested SZX for the COAP block request
+ * @param[in]   work_buf   Work buffer, must be `NANOCOAP_BLOCKWISE_BUF(blksize)` bytes
+ * @param[in]   callback   callback to be executed on each received block
+ * @param[in]   arg        optional function arguments
+ *
+ * @returns     -EINVAL    if an invalid url is provided
+ * @returns     -1         if failed to fetch the url content
+ * @returns      0         on success
+ */
+int nanocoap_get_blockwise_url(const char *url,
+                               coap_blksize_t blksize, void *work_buf,
+                               coap_blockwise_cb_t callback, void *arg);
+
+/**
+ * @brief   Simple synchronous CoAP request
+ *
+ * @param[in]       sock    socket to use for the request
+ * @param[in,out]   pkt     Packet struct containing the request. Is reused for
+ *                          the response
+ * @param[in]       len     Total length of the buffer associated with the
+ *                          request
+ *
+ * @returns     length of response on success
+ * @returns     <0 on error
+ */
+ssize_t nanocoap_request(sock_udp_t *sock, coap_pkt_t *pkt, size_t len);
```

Comment:
I can change this to `nanocoap_sock_request()` so we can leave the old API in place